specification/k3po.control/SPEC.md

```diff
@@ -240,3 +217,50 @@ This event is sent by the driver when the driver encountered an error while proc
 The message body for the event is the description of the exception in case we have an exception or a more detailed description in case
 there is no exception. 
 
+
+## Message Exchanges
+
+The test framework initializes a connection to the driver.  Only one test may be in execution at a time per a single 
+connection.  
+
+The test framework initiates the start of a test with a **PREPARE** command. When receiving a **PREPARE** the driver SHOULD 
+load and prepare a set of scripts for execution.   If there are `accept` instructions in the scripts, a bind on 
+corresponding ports/files MUST be performed during preparation.
+
+When the scripts are successfully prepared, the driver MUST send a **PREPARED** event.
+
+The test framework signals the start of a test with the **START** command.  The driver MUST start the script execution
+when it receives the **START** command.
+
+The driver MUST send the **STARTED** event when the scripts have started.
+
+The driver MUST send the **FINISHED** event, when
+  1) all K3PO scripts have finished, either by successfully completing or 
+hitting undefined behavior AND 
+  2) after sending a **STARTED** event (note, script execution may have completed if the script
+is just accepts prior to the **START** command or **STARTED** event)
+
```

Comment:
> The ABORT case is not clearly specified here. I see it bellow, but it can be missed if you don't read carefully the entire paragraph.

I moved that paragraph closer.

> Also, I don't see anywhere the fact that FINISHED is sent maximum one time. In case of an exception running the script we might not even have a FINISHED event, but an ERROR.

Is that not implied?  "Error" is unrecoverable, and "FINISHED" is sent when script execution stops,  that can only be once, correct?